notebook/services/contents/tests/test_manager.py

```diff
@@ -528,3 +528,34 @@ def test_check_and_sign(self):
         cm.mark_trusted_cells(nb, path)
         cm.check_and_sign(nb, path)
         assert cm.notary.check_signature(nb)
+
+
+class TestContentsManagerNoAtomic(TestCase):
+
+    def setUp(self):
+        self._temp_dir = TemporaryDirectory()
+        self.td = self._temp_dir.name
+        self.contents_manager = FileContentsManager(
+            root_dir = self.td,
+        )
+        self.contents_manager.use_atomic_writing = False
+
+    def tearDown(self):
+        self._temp_dir.cleanup()
+
+    def new_notebook_no_atomic(self):
+        """
+
+        """
+        cm = self.contents_manager
+        model = cm.new_untitled(type="notebook")
+        name = model['name']
+        path = model['path']
+
+        full_model = cm.get(path)
+        nb = full_model['content']
+        nb['metadata']['counter'] = int(1e6 * time.time())
+        self.add_code_cell(nb)
+
+        cm.save(full_model, path)
+        return nb, name, path
```

Comment:
I am not sure my new solution would please you, as I relaunch the exact same set of test in non atomic writing case than in atomic writing case (this time I checked the tests are eventally done).
But I am sure of the exact way the orignal set of test check the presence of file on disk after a save.